File: java/src/test/java/io/binac/leetcode/FindFirstAndLastPositionOfElementInSortedArrayTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class FindFirstAndLastPositionOfElementInSortedArrayTest {
    private static final FindFirstAndLastPositionOfElementInSortedArray.Solution1 SOLUTION1 = new FindFirstAndLastPositionOfElementInSortedArray.Solution1();

    private final int[] NUMS1 = {5, 7, 7, 8, 8, 10};
    private final int TARGET1 = 8;
    private final int[] EXPECTED1 = {3, 4};

    private final int[] NUMS2 = {5, 7, 7, 8, 8, 10};
    private final int TARGET2 = 6;
    private final int[] EXPECTED2 = {-1, -1};

    private final int[] NUMS3 = {};
    private final int TARGET3 = 0;
    private final int[] EXPECTED3 = {-1, -1};

    @Test
    void test1() {
        assertArrayEquals(EXPECTED1, SOLUTION1.searchRange(NUMS1, TARGET1));
        assertArrayEquals(EXPECTED2, SOLUTION1.searchRange(NUMS2, TARGET2));
        assertArrayEquals(EXPECTED3, SOLUTION1.searchRange(NUMS3, TARGET3));
    }
}

File: cpp/tests/FindFirstAndLastPositionOfElementInSortedArrayTest.cpp
#include "FindFirstAndLastPositionOfElementInSortedArray.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(FindFirstAndLastPositionOfElementInSortedArray, Solution34_1) {
  auto S34_1 = Solution34_1();

  std::vector<int> Nums1{5, 7, 7, 8, 8, 10};
  int const Target1 = 8;
  std::vector<int> const Expected1{3, 4};
  EXPECT_EQ(Expected1, S34_1.searchRange(Nums1, Target1));

  std::vector<int> Nums2{5, 7, 7, 8, 8, 10};
  int const Target2 = 6;
  std::vector<int> const Expected2{-1, -1};
  EXPECT_EQ(Expected2, S34_1.searchRange(Nums2, Target2));

  std::vector<int> Nums3;
  int const Target3 = 0;
  std::vector<int> const Expected3{-1, -1};
  EXPECT_EQ(Expected3, S34_1.searchRange(Nums3, Target3));
}
